File: backend/src/main/java/com/miniprogram/entity/MembershipPlan.java
package com.miniprogram.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 付费会员档位（平台/星球），与积分成长 {@link MemberLevel} 分离。
 */
@Data
@TableName(value = "mp_membership_plan", autoResultMap = true)
public class MembershipPlan implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId
    private Long id;

    /** platform | planet */
    private String scope;

    /** scope=planet 时必填；platform 必须为 null */
    private String planetId;

    private String name;

    private String icon;

    private String description;

    /** 权益码列表 */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> rights;

    /** 仅平台档用于商城折扣 */
    private BigDecimal discountRate;

    /** 仅平台档：赠送目标星球 */
    private String giftPlanetId;

    /** 仅平台档：赠送天数；0 = 不赠送 */
    private Integer giftPlanetDays;

    /** 评论区/星球显示会员角标：1=开 0=关 */
    private Integer showBadge;

    /** 到期前提醒天数；0 = 关闭 */
    private Integer expireRemindDays;

    private Integer sortOrder;

    /** 1 启用 / 0 禁用 */
    private Integer status;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField("created_at")
    private LocalDateTime createTime;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField("updated_at")
    private LocalDateTime updateTime;
}

File: admin/src/api/membershipPlan.ts
/**
 * 付费会员档 API（平台 / 星球）
 */
import { get, post, put, del } from './request'

const BASE_URL = '/api/v1/admin/membership-plans'

export type MembershipPlanScope = 'platform' | 'planet'

export interface MembershipPlan {
  id: number
  scope: MembershipPlanScope
  planetId?: string | null
  name: string
  icon?: string
  description?: string
  rights?: string[]
  discountRate?: number | null
  giftPlanetId?: string | null
  giftPlanetDays?: number
  /** 1=显示会员角标 */
  showBadge?: number
  /** 到期前提醒天数；0=关闭 */
  expireRemindDays?: number
  sortOrder?: number
  status: number
  createdAt?: string
  updatedAt?: string
}

export interface MembershipPlanPayload {
  scope: MembershipPlanScope
  planetId?: string | null
  name: string
  icon?: string
  description?: string
  rights?: string[]
  discountRate?: number | null
  giftPlanetId?: string | null
  giftPlanetDays?: number
  showBadge?: number
  expireRemindDays?: number
  sortOrder?: number
  status?: number
}

function normalizePlan(row: any): MembershipPlan {
  return {
    id: Number(row.id),
    scope: (row.scope || 'platform') as MembershipPlanScope,
    planetId: row.planetId ?? row.planet_id ?? null,
    name: row.name || '',
    icon: row.icon,
    description: row.description,
    rights: Array.isArray(row.rights) ? row.rights.map(String) : [],
    discountRate: row.discountRate ?? row.discount_rate ?? null,
    giftPlanetId: row.giftPlanetId ?? row.gift_planet_id ?? null,
    giftPlanetDays: Number(row.giftPlanetDays ?? row.gift_planet_days ?? 0),
    showBadge: Number(row.showBadge ?? row.show_badge ?? 0),
    expireRemindDays: Number(row.expireRemindDays ?? row.expire_remind_days ?? 0),
    sortOrder: Number(row.sortOrder ?? row.sort_order ?? 0),
    status: Number(row.status ?? 1),
    createdAt: row.createdAt ?? row.created_at,
    updatedAt: row.updatedAt ?? row.updated_at,
  }
}

export function getMembershipPlanList(params?: { scope?: MembershipPlanScope; planetId?: string }) {
  return get<any[]>(BASE_URL, params as Record<string, unknown>).then((res: any) => ({
    ...res,
    data: (res.data || []).map(normalizePlan),
  }))
}

export function createMembershipPlan(data: MembershipPlanPayload) {
  return post<any>(BASE_URL, data).then((res: any) => ({
    ...res,
    data: normalizePlan(res.data),
  }))
}

export function updateMembershipPlan(id: number, data: MembershipPlanPayload) {
  return put<any>(`${BASE_URL}/${id}`, data).then((res: any) => ({
    ...res,
    data: normalizePlan(res.data),
  }))
}

export function deleteMembershipPlan(id: number) {
  return del<void>(`${BASE_URL}/${id}`)
}
